### PR TITLE
Fix losing keys on PrivateSend

### PR DESCRIPTION
**Pull request by UdjinM6**
Looks like #1567 is abandoned. Continuing from where it stopped and addressing my own comments.

### CODE REVIEW

**Comment by krychlicki** [comment]
Apologies for the delay in implementing changes!
Please see my comments below.

File: src/privatesend-util.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "wallet/wallet.h"
+
+class CKeyHolder
+{
+private:
+    CReserveKey reserveKey;
+    CPubKey pubKey;
+public:
+    CKeyHolder(CWallet* pwalletIn);
```
Comment (krychlicki) [inline src/privatesend-util.h]: Disable copy/assign and allow move by using:
CKeyHolder(CKeyHolder&&) = default;
CKeyHolder& operator=(CKeyHolder&&) = default;

File: src/privatesend-client.cpp [hunk 2]
```diff
@@ -1261,24 +1272,23 @@ bool CPrivateSendClient::CreateDenominated(const CompactTallyItem& tallyItem, bo
...
     // make our collateral address
-    CReserveKey reservekeyCollateral(pwalletMain);
-
-    CScript scriptCollateral;
-    CPubKey vchPubKey;
-    assert(reservekeyCollateral.GetReservedKey(vchPubKey, false)); // should never fail, as we just unlocked
-    scriptCollateral = GetScriptForDestination(vchPubKey.GetID());
-
-    // ****** Add collateral outputs ************ /
+    CReserveKey reservekeyCollateral(pwalletMain);
```
Comment (krychlicki) [inline src/privatesend-client.cpp]: reservekeyCollateral can be also stored in:
CKeyHolderStorage keyHolderStorageDenom

**Comment by UdjinM6** [comment]
fixed

**Comment by krychlicki** [comment]
utACK

**Pull request merged**